examples/rust/custom_space_view/src/color_coordinates_view_part_system.rs

```diff
@@ -21,13 +21,66 @@ pub struct ColorWithInstanceKey {
     pub instance_key: InstanceKey,
 }
 
+struct ColorArchetype;
+
+// TODO(#2778): The introduction of ArchetypeInfo should make this much much nicer.
+#[allow(clippy::unimplemented)]
+impl re_types::Archetype for ColorArchetype {
+    fn name() -> re_types::ArchetypeName {
+        "InstanceColor".into()
+    }
+
+    fn required_components() -> &'static [re_types::ComponentName] {
+        static COMPONENTS: once_cell::sync::Lazy<[re_types::ComponentName; 1]> =
+            once_cell::sync::Lazy::new(|| [re_types::components::Color::name()]);
+        &*COMPONENTS
+    }
+
+    fn recommended_components() -> &'static [re_types::ComponentName] {
+        &[]
+    }
+
+    fn optional_components() -> &'static [re_types::ComponentName] {
+        &[]
+    }
+
+    fn all_components() -> &'static [re_types::ComponentName] {
+        Self::required_components()
+    }
+
+    fn try_to_arrow(
+        &self,
+    ) -> re_types::SerializationResult<
+        Vec<(
+            re_viewer::external::arrow2::datatypes::Field,
+            Box<dyn re_viewer::external::arrow2::array::Array>,
+        )>,
+    > {
+        unimplemented!()
+    }
+
+    fn try_from_arrow(
+        _data: impl IntoIterator<
+            Item = (
+                re_viewer::external::arrow2::datatypes::Field,
+                Box<dyn re_viewer::external::arrow2::array::Array>,
+            ),
+        >,
+    ) -> re_viewer::external::re_types::DeserializationResult<Self>
+    where
+        Self: Sized,
+    {
+        unimplemented!()
+    }
```

Comment:
if we were able to compress the visual space of this somehow it wouldn't be too bad 🤔